be/example/src/app.ts

```diff
@@ -1,9 +1,32 @@
+import '../../src/register';
+
+import { RequestListener } from 'http';
+
 import Koa from 'koa';
 
 import { createMiddleware } from './api';
+import { PORT } from './config';
 
-export const createApp = async () => {
+const createApp = async () => {
   const middleware = await createMiddleware();
 
   return new Koa().use(middleware);
 };
+
+const wrapAsyncApp = (appPromise: Promise<Koa>) => {
+  const callbackPromise = appPromise.then((app) => app.callback());
+
+  return {
+    callback: (): RequestListener => (req, res) =>
+      callbackPromise.then((callback) => callback(req, res)),
+
+    listen: (port?: number, listeningListener?: () => void) =>
+      appPromise.then((app) => app.listen(port, listeningListener)),
+
+    port: PORT,
+  };
+};
```

Comment:
This little dance is pretty gross and annoying. I'm wondering if we should bundle it in Koala?

Reply:
Just realised I was overcomplicating this for no reason. I'll still package it up as a Koala proposal and see where it goes.